x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.ts

```diff
@@ -54,39 +55,48 @@ export const singleBulkIndex = async (
     },
     buildBulkBody(doc, params),
   ]);
+  const time1 = performance.now();
   const firstResult = await service.callCluster('bulk', {
-    refresh: true,
+    index: process.env.SIGNALS_INDEX || '.siem-signals-10-01-2019',
+    refresh: false,
     body: bulkBody,
   });
+  const time2 = performance.now();
+  logger.info(`individual bulk process time took: ${time2 - time1} milliseconds`);
+  logger.info(`took property says bulk took: ${firstResult.took} milliseconds`);
```

Comment:
Careful here. The `service.callCluster` looks like it is typed with the dreaded `any`.

<img width="620" alt="Screen Shot 2019-10-29 at 1 56 20 PM" src="https://user-images.githubusercontent.com/1151048/67804128-fb7ccb00-fa53-11e9-9f32-a1cf13c197ce.png">

Which means this `firstResult` can contain anything and typescript doesn't care what you attempt to call on the object:

<img width="483" alt="Screen Shot 2019-10-29 at 1 56 13 PM" src="https://user-images.githubusercontent.com/1151048/67804161-0d5e6e00-fa54-11e9-9e67-ca86274dd2d6.png">

No changes if you trust it and don't want to ... or what I typically do is add a typing and then do a cast right when it comes out.

I don't know what the shape of this source hit is but you can do an unknown like so:

```ts
const firstResult: SearchResponse<unknown> = await service.callCluster('bulk', {
```

If you don't care or know. If it's a signal search it then it would be:

```ts
const firstResult: SignalSearchResponse = await service.callCluster('bulk', {
```

or you can follow what `SignalSearchResponse` looks like and add your own source type.

Regardless of choice one missing type we have upstream is the optional `error` you're suing so you would also add that to our SearchResponse generic from here:
x-pack/legacy/plugins/siem/server/lib/types.ts

on line 116

like so:

<img width="552" alt="Screen Shot 2019-10-29 at 2 03 28 PM" src="https://user-images.githubusercontent.com/1151048/67804830-5c58d300-fa55-11e9-9868-526f9fe32da4.png">

Reply:
You can type the `errors` more, but it's better to use a generic object or an unknown to start with and others as needed will add more typings.

The `any` patterns in that typing are not what you want to do. Some types are so large its too overwhelming to add all of them at once but doing typings "just in time" and "as needed" slowly over time gives us more and more protection against changes. So just use your discretion and tolerance for how much time to spend.